securedrop/journalist.py

```diff
@@ -430,15 +598,85 @@ def index():
             Submission.query.filter_by(source_id=source.id,
                                        downloaded=False).all())
 
-    return render_template('index.html', unstarred=unstarred, starred=starred)
+    return starred, unstarred, source_labels, all_source_labels
+
+
+@app.route('/')
+@login_required
+def index():
+    # Long SQLAlchemy statements look best when formatted according to
+    # the Pocoo style guide, IMHO:
+    # http://www.pocoo.org/internal/styleguide/
+    sources = Source.query.filter_by(pending=False) \
+                          .order_by(Source.last_updated.desc()) \
+                          .all()
+
+    starred, unstarred, labels, all_labels = get_stars_and_labels(sources)
+
+    return render_template('index.html', unstarred=unstarred, starred=starred,
+                           source_labels=labels, all_labels=all_labels)
+
+
+@app.route('/filter_by_label/<int:label_id>', methods=('POST',))
```

Comment:
We discussed this functionality and you mentioned that it is currently only possible to filter by a single SourceTag at a time. It would be nice if you could filter by an arbitrary number of tags, which would be AND'ed together.